### PR TITLE
Lps 202517 AI Generate images - add forms and mockup implementation

### PR DESCRIPTION
**Pull request by robertoDiaz**
**Issues:**
https://liferay.atlassian.net/browse/LPS-202516 Add the new form to get Images from Dall-e
https://liferay.atlassian.net/browse/LPS-202517 Create the backend logic to call the client (by now the mock client with testing porpouses)
https://liferay.atlassian.net/browse/LPS-202518 Add the generated image selection screen

**Motivation**
This pull allows to generate from 1 to 4 images via Dalle API. Tests have been done using a mock client.

In future pulls we will save the selected images, and format them via selectable cards.

**How to test it:**
See https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2413429012/How+to+use+the+OpenAI+mock+client for reference.

1. Enable mock Client via Gogo shell (`scr:enable com.liferay.ai.creator.openai.web.internal.client.MockAICreatorOpenAIClient)`
2. Set VALID_API_KEY as Apy key for AI Creator in Instance Settings
3. Go to D&M and click Create AI Image (from the + dropdown)
4. On Description, add USER_IMAGES_URL_ + an url that exists. i.e `USER_IMAGES_URL_https://fastly.picsum.photos/id/1/5000/3333.jpg?hmac=Asv2DU3rA_5D1xSe22xZK47WEAN0wjWeFOhzd13ujW4`
AIScreeshot

### CODE REVIEW

**Comment by robertoDiaz** [comment]
@ambrinchaudhary 

Please rebase

**Comment by ambrinchaudhary** [comment]
ci:test:sf

**Review by AliciaGarciaGarcia — CHANGES_REQUESTED** [review]
There is an strange movement (on my local) when you save the instance configuration:

https://github.com/liferay-lima/liferay-portal/assets/47524751/6d528fb0-bba6-4ae6-a174-1db77e5d967a

**Comment by robertoDiaz** [comment]
Hi @AliciaGarciaGarcia 

> Why was decided to enable the configuration by default?

This is not part of this PR.

Anyway, to provide more contenxt, this is due consistency with previous properties (that were enabled by default).

About https://github.com/liferay-lima/liferay-portal/pull/4965#pullrequestreview-1757049388

We already seen that in the previous stories, but seems related with frontend infrastructure (see that the user icon also changes)

File: modules/apps/ai-creator/ai-creator-openai-web/src/main/resources/META-INF/resources/view.jsp [hunk 1]
```diff
@@ -16,8 +16,18 @@ AICreatorOpenAIDisplayContext aiCreatorOpenAIDisplayContext = (AICreatorOpenAIDi
...
 
-	<react:component
-		module="ai_creator_modal/AICreatorModal"
-		props="<%= aiCreatorOpenAIDisplayContext.getProps() %>"
-	/>
+	<c:choose>
+		<c:when test="<%= aiCreatorOpenAIDisplayContext.isGetGenerations() %>">
+			<react:component
+				module="ai_creator_modal/AICreatorImageModal"
+				props="<%= aiCreatorOpenAIDisplayContext.getProps() %>"
```
[the remark this answers is not in the document]
Comment (robertoDiaz) [reply modules/apps/ai-creator/ai-creator-openai-web/src/main/resources/META-INF/resources/view.jsp:23]: I'd prefer to keep it in a single method, but if you prefer I could add it.

Please confirm if this is an optional change and if not, I'll do it.

Regards

[the remark this answers is not in the document]
Comment (robertoDiaz) [reply modules/apps/ai-creator/ai-creator-openai-web/src/main/resources/META-INF/resources/view.jsp:23]: @ambrinchaudhary also agrees with your opinion.

Democracy has speak :)

**Review by AliciaGarciaGarcia — APPROVED** [review]

**Comment by AliciaGarciaGarcia** [comment]
ci:forward

**Comment by Jeremy-Ch** [comment]
The failures are unrelated.

**Comment by Jeremy-Ch** [comment]
ci:forward:force

**Pull request closed without merging**